agent/consul/internal_endpoint.go

```diff
@@ -85,8 +86,22 @@ func (m *Internal) EventFire(args *structs.EventFireRequest,
 	// Add the consul prefix to the event name
 	eventName := userEventName(args.Name)
 
-	// Fire the event
-	return m.srv.serfLAN.UserEvent(eventName, args.Payload, false)
+	// Fire the event on all LAN segments
+	segments := m.srv.LANSegments()
```

Comment:
How come you decided not to include the default segment in `LANSegments()`, which would eliminate the special cases? I think you had that way partway through development.

Reply:
Yeah, I removed it from the segments map on the server but I should have re-added it to the map that `LANSegments()` returns, fixed